src/app/pages/ArticlePage/index.test.tsx

```diff
@@ -354,6 +354,102 @@ describe('Article Page', () => {
+      includeJumpTo: boolean;
+      includeRelatedContent: boolean;
+    }) => {
+      let blocks = [...articleDataPidgin.content.model.blocks];
+
+      if (includeJumpTo) {
+        blocks = [
+          ...blocks,
+          {
+            id: 'jumpTo',
+            type: 'jumpTo',
+            model: {
+              jumpToHeadings: [
+                { heading: 'Harris separates from Biden' },
+                { heading: 'Prison gender surgery debate' },
+                { heading: 'Apology challenge' },
+                { heading: "Biden's mental state questioned" },
+              ],
+            },
+          },
+        ];
+      }
+
+      if (includeRelatedContent) {
+        blocks = [
+          ...blocks,
+          {
+            id: '37b8a761',
+            type: 'relatedContent',
+            model: {},
+          },
+        ];
+      }
+
+      return {
+        ...articleDataPidgin,
+        content: {
+          ...articleDataPidgin.content,
+          model: {
+            ...articleDataPidgin.content.model,
+            blocks,
+          },
+        },
+      };
+    };
+
+    it.each`
+      description                                                                                       | includeJumpTo | includeRelatedContent | expectedJumpTo | expectedRelatedContent
+      ${'should render a jumpTo component if jumpToHeadings are present'}                               | ${true}       | ${false}              | ${true}        | ${false}
+      ${'should not render a jumpTo component if jumpToHeadings are not present'}                       | ${false}      | ${false}              | ${false}       | ${false}
+      ${'should render a jumpTo component with relatedContent link if relatedContent block is present'} | ${true}       | ${true}               | ${true}        | ${true}
+    `(
+      '$description',
+      async ({
+        includeJumpTo,
+        includeRelatedContent,
+        expectedJumpTo,
+        expectedRelatedContent,
+      }) => {
+        const pageData = addJumpToAndRelatedContentToPageData({
+          includeJumpTo,
+          includeRelatedContent,
+        });
+
+        const { container } = render(
+          <Context service="pidgin">
+            <ArticlePage pageData={pageData} />
+          </Context>,
+          { service: 'pidgin' },
+        );
+        await waitFor(() => {
+          const jumpToSection = container.querySelector(
+            '[data-testid="jump-to"]',
+          );
+          if (expectedJumpTo) {
+            expect(jumpToSection).not.toBeNull();
+            if (expectedRelatedContent) {
+              const relatedContentLink = jumpToSection?.querySelector(
+                '[data-testid="jump-to-link-related-content-heading"]',
+              );
+              expect(relatedContentLink).not.toBeNull();
+            }
+          } else {
+            expect(jumpToSection).toBeNull();
+          }
+        });
+
+        expect(container).toMatchSnapshot();
```

Comment:
Could we try to avoid snapshotting the entire article container - for the purposes of the test it's enough to check if the component renders or not, depending on the scenario, and using the data-testid query selector is perfectly adequate. 

Let's remove this

```suggestion
```

Reply:
snapshot removed